repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -55,519 +54,363 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
-              case Failure(e: UnparseableTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-              case Failure(e: IdentityCommunicationException) => Reject(SC_BAD_GATEWAY, Some(e.getMessage))
-              case Failure(e: UnauthorizedEndpointException) => Reject(SC_FORBIDDEN, Some(e.getMessage))
-              case Failure(e: OverLimitException) =>
-                response.addHeader(HttpHeaders.RETRY_AFTER, e.retryAfter)
-                if (isSelfValidating) {
-                  Reject(e.statusCode, Some(e.getMessage))
-                } else {
-                  Reject(SC_SERVICE_UNAVAILABLE, Some(e.getMessage))
-                }
-              case Failure(e) if e.getCause.isInstanceOf[AkkaServiceClientException] && e.getCause.getCause.isInstanceOf[TimeoutException] =>
-                Reject(SC_GATEWAY_TIMEOUT, Some(s"Call timed out: ${e.getMessage}"))
-              case Failure(_: AdminTokenUnauthorizedException) if isSelfValidating =>
-                Reject(SC_UNAUTHORIZED, Some("Token unauthorized"))
-              case Failure(e) => Reject(SC_INTERNAL_SERVER_ERROR, Some(e.getMessage))
-            }
-          }
+    response.uncommit()
+    if (response.getStatus == SC_UNAUTHORIZED) {
+      response.addHeader(HttpHeaders.WWW_AUTHENTICATE, s"Keystone uri=${configuration.getIdentityService.getUri}")
+    }
+    response.commitToResponse()
+  }
 
-        filterResult match {
-          case Pass =>
-            logger.trace("Processing completed, passing to next filter or service")
-            addIdentityStatusHeader(confirmed = true)
-            chain.doFilter(request, response)
-          case Reject(statusCode, message) =>
-            Option(config.getDelegating) match {
-              case Some(delegating) =>
-                logger.debug(s"Delegating with status $statusCode caused by: ${message.getOrElse("unspecified")}")
-
-                val delegationHeaders = buildDelegationHeaders(statusCode,
-                  "keystone-v2",
-                  message.getOrElse("Failure in the Keystone v2 filter").replace("\n", " "),
-                  delegating.getQuality)
-
-                addIdentityStatusHeader(confirmed = false)
-                delegationHeaders foreach { case (key, values) =>
-                  values foreach { value =>
-                    request.addHeader(key, value)
-                  }
-                }
+  override def doAuth(request: HttpServletRequestWrapper): Try[Unit.type] = {
+    /**
+      * DECLARE COMMON VALUES
+      */
+    lazy val traceId = Option(request.getHeader(CommonHttpHeader.TRACE_GUID)).filter(_ => sendTraceHeader)
+    lazy val requestHandler = new KeystoneRequestHandler(configuration.getIdentityService.getUri, akkaServiceClient, traceId)
 
-                chain.doFilter(request, response)
+    /**
+      * DEFINING FUNCTIONS IN SCOPE
+      */
+    def isWhitelisted(requestUri: String): Boolean = {
```

Comment:
Hm, this shouldn't be here anymore, should it? It looks like it was moved to the abstract filter.